Apply style guide (#2916)
modelica/ModelicaSpecification: chapters/statements.tex

\subsection{If-Statement}\label{if-statement}

> \lstinline!if!-statements\index{if@\robustinline{if}!statement}\index{then@\robustinline{then}!if-statement@\robustinline{if}-statement}\index{else@\robustinline{else}!if-statement@\robustinline{if}-statement}\index{elseif@\robustinline{elseif}!if-statement@\robustinline{if}-statement} have the following syntax:

Reviewer: ```suggestion
The \lstinline!if!-statements\index{if@\robustinline{if}!statement}\index{then@\robustinline{then}!if-statement@\robustinline{if}-statement}\index{else@\robustinline{else}!if-statement@\robustinline{if}-statement}\index{elseif@\robustinline{elseif}!if-statement@\robustinline{if}-statement} have the following syntax:
```

  y2 = sin(y1);
algorithm
  when x > 2 then
    y3 := 2 * x + y1 + y2;
  end when;
\end{lstlisting}
Merging the \lstinline!when!-statements can lead to less efficient code and different models with different behavior depending on the order of the assignment to \lstinline!y1! and \lstinline!y3! in the algorithm.
\end{example}

\subsubsection{Restrictions on Where a When-statement may occur}\label{restrictions-on-where-a-when-statement-may-occur}

\begin{itemize}
\item
  A \lstinline!when!-statement shall not be used within a function.
\item

>   \lstinline!when!-statements shall not occur inside initial algorithms.

Reviewer: To be similar to first item.
```suggestion
  A \lstinline!when!-statement shall not occur inside an initial algorithm.
```

\item

>   \lstinline!when!-statements cannot be nested.

Reviewer: Unsure, but could write:
```suggestion
  A \lstinline!when!-statement cannot be nested inside another \lstinline!when!-statement.
```

\item

>   \lstinline!when!-statements shall not occur inside \lstinline!while!-loops, \lstinline!for!-loops, or \lstinline!if!-statements in algorithms.

Reviewer: ```suggestion
 A \lstinline!when!-statements shall not occur inside \lstinline!while!-loops, \lstinline!for!-loops, or \lstinline!if!-statements in an algorithm.
```

\subsubsection{Restrictions on Statements within When-Statements}\label{restrictions-on-statements-within-when-statements}

\begin{nonnormative}
In contrast to \lstinline!when!-equations, \cref{restrictions-on-equations-within-when-equations}, there are no additional restrictions within \lstinline!when!-statements:
\begin{itemize}
\item
  In algorithms, all assignment statements are already restricted to left-hand-side variables.
\item
  If at least one element of an array appears on the left-hand-side of the assignment operator inside a \lstinline!when!-statement, it is as if the entire array appears in the left-hand-side according to \cref{execution-of-an-algorithm-in-a-model}.
  Thus, there is no need to restrict the indices to parameter-expressions.
\item

>   \lstinline!for!-loops and \lstinline!if!-statements are not problematic inside \lstinline!when!-statements in algorithms, since all left-hand-side variables inside \lstinline!when!-statements are assigned to their pre-values before the start of the algorithm, according to \cref{execution-of-an-algorithm-in-a-model}.

Reviewer: ```suggestion
  The \lstinline!for!-loops and \lstinline!if!-statements are not problematic inside \lstinline!when!-statements in algorithms, since all left-hand-side variables inside \lstinline!when!-statements are assigned to their pre-values before the start of the algorithm, according to \cref{execution-of-an-algorithm-in-a-model}.
```
